bluesky/run_engine.py

```diff
@@ -656,9 +658,18 @@ def _check_for_signals(self):
         # Check for pause requests from keyboard.
         if self.state.is_running:
             if self._sigint_handler.interrupted:
-                self.debug("RunEngine detected a SIGINT (Ctrl+C)")
-                loop.call_soon(self.request_pause, False, 'SIGINT')
                 self._sigint_handler.interrupted = False
+                with SignalHandler(signal.SIGINT) as second_sigint_handler:
+                    self.debug("RunEngine detected a SIGINT (Ctrl+C)")
+                    # We know we will either pause or abort, so we can
+                    # hold up the scan now....
+                    ttime.sleep(0.5)  # give user time to hit Ctrl+C again
```

Comment:
does this eat _all_ sigint for half a second?  Might be better to do this as a loop over a shorter interval